engine/src/main/java/org/pentaho/di/repository/filerep/KettleFileRepository.java

```diff
@@ -0,0 +1,1571 @@
+  }
+
+  public ObjectId[] getTransformationClusterSchemaIDs( ObjectId id_transformation ) throws KettleException {
+    return new ObjectId[] {};
+  }
+
+  @Override
+  public ObjectId getTransformationID( String name, RepositoryDirectoryInterface repositoryDirectory ) throws KettleException {
+    return getObjectId( repositoryDirectory, name, EXT_TRANSFORMATION );
+  }
+
+  @Override
+  public String[] getTransformationNames( ObjectId id_directory, boolean includeDeleted ) throws KettleException {
+    try {
+      List<String> list = new ArrayList<String>();
+
+      RepositoryDirectoryInterface tree = loadRepositoryDirectoryTree();
+      RepositoryDirectoryInterface directory = tree.findDirectory( id_directory );
+
+      String folderName = calcDirectoryName( directory );
+      FileObject folder = KettleVFS.getFileObject( folderName );
+
+      for ( FileObject child : folder.getChildren() ) {
+        if ( child.getType().equals( FileType.FILE ) ) {
+          if ( !child.isHidden() || !repositoryMeta.isHidingHiddenFiles() ) {
+            String name = child.getName().getBaseName();
+
+            if ( name.endsWith( EXT_TRANSFORMATION ) ) {
+
+              String transName = name.substring( 0, name.length() - 4 );
+              list.add( transName );
+            }
+          }
+        }
+      }
+
+      return list.toArray( new String[list.size()] );
+    } catch ( Exception e ) {
+      throw new KettleException(
+        "Unable to get list of transformations names in folder with id : " + id_directory, e );
+    }
+  }
+
+  public ObjectId[] getTransformationPartitionSchemaIDs( ObjectId id_transformation ) throws KettleException {
+    return new ObjectId[] {};
+  }
+
+  public String[] getTransformationsUsingCluster( ObjectId id_cluster ) throws KettleException {
+    return new String[] {};
+  }
+
+  @Override
+  public String[] getTransformationsUsingDatabase( ObjectId id_database ) throws KettleException {
+    return new String[] {};
+  }
+
+  public String[] getTransformationsUsingPartitionSchema( ObjectId id_partition_schema ) throws KettleException {
+    return new String[] {};
+  }
+
+  public String[] getTransformationsUsingSlave( ObjectId id_slave ) throws KettleException {
+    return new String[] {};
+  }
+
+  @Override
+  public String getVersion() {
+    return FILE_REPOSITORY_VERSION;
+  }
+
+  public ObjectId insertClusterSlave( ClusterSchema clusterSchema, SlaveServer slaveServer ) throws KettleException {
+    return null;
+  }
+
+  @Override
+  public void insertJobEntryDatabase( ObjectId id_job, ObjectId id_jobentry, ObjectId id_database ) throws KettleException {
+  }
+
+  public void insertJobNote( ObjectId id_job, ObjectId id_note ) throws KettleException {
+  }
+
+  @Override
+  public ObjectId insertLogEntry( String description ) throws KettleException {
+    String logfile = calcDirectoryName( null ) + LOG_FILE;
+    try {
+      OutputStream outputStream = KettleVFS.getOutputStream( logfile, true );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *outputStream*. The resource is referenced at lines 841, 842. The resource is closed at line 843. There are other execution paths that do not contain closure statements, for example, when OutputStream.write throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).